### PR TITLE
Roosa: Reference tree part added

### PR DESCRIPTION
**Pull request by ifosch**
Tests coming in following PR

### CODE REVIEW

**Comment by dsager** [comment]
+1
eew, tabs :)

File: mcc/roosa/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+	if zoneName == "" {
+		log.Fatal("Insufficient input parameters!")
+	}
+}
+
+// PrintNode prints a Node, and all its children.
+func PrintNode(node *Node, indent int) {
+	indents := ""
+	for i := 0; i <= indent-1; i++ {
```
Comment (miguelbernadi) [inline mcc/roosa/cmd/main.go]: Use < instead of <= and you no longer need indent - 1

File: mcc/roosa/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		indents += "\t"
+	}
+	extra := ""
+	for i, record := range node.content.ResourceRecords {
```
Comment (miguelbernadi) [inline mcc/roosa/cmd/main.go]: Can't you just do:
`for i, record := range node.content.ResourceRecords[:-1] {`
That would avoid the if altogether.

File: mcc/roosa/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	}
+	fmt.Printf("%v%v %v %v\n", indents, *node.content.Name, *node.content.Type, extra)
+	indent++
+	for _, child := range node.children {
+		PrintNode(child, indent)
+	}
+}
+
+// FilterReferrals returns 'A', 'AAAA', and 'CNAME' records from a []*route53.ResourceRecordSet.
+func FilterReferrals(records []*route53.ResourceRecordSet) (filteredRecords []*route53.ResourceRecordSet) {
```
Comment (miguelbernadi) [inline mcc/roosa/cmd/main.go]: If you don't name the output, you can just return the internal function's result.

File: mcc/roosa/cmd/main.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+		[]string{
+			"A",
+			"AAAA",
+			"CNAME",
+		},
+	)
+	return
+}
+
+// FillReferenceTrees fills a referral lookup table with the base records form a []*route53.ResourceRecordSet.
```
Comment (miguelbernadi) [inline mcc/roosa/cmd/main.go]: *from

**Comment by ifosch** [comment]
@miguelbernadi fixed

**Comment by miguelbernadi** [comment]
+1

**Pull request merged**